pkg/framework/nodes.go

```diff
@@ -71,7 +72,11 @@ func GetNodesFromMachineSet(client runtimeclient.Client, machineSet *mapiv1beta1
 	var nodes []*corev1.Node
 	for key := range machines {
 		node, err := GetNodeForMachine(client, machines[key])
-		if err != nil {
+		if err != nil && apierrors.IsNotFound(err) {
+			// We don't care about not found errors.
+			// Callers should account for the number of nodes being correct or not.
```

Comment:
It still will be better to put a log message in there. The method signature suggests it should be a matching list to the assigned Machines.
```golang
klog.Infof("No Node object found for machine %s", machine.GetName())
```